subsys/metrics/src/main/java/org/commonjava/indy/metrics/zabbix/sender/IndyZabbixSender.java

```diff
@@ -247,200 +230,96 @@ private IndyZabbixSender( boolean bCreateNotExistHostGroup, boolean bCreateNotEx
+                if ( hostid != null )
+                {
                     hostCache.put( host, hostid );
 
-                    logger.info( "!result.isEmpty() " + hostid );
                 }
                 else
-                {
-                    String hostId = createHost( host, ip );
-                    hostCache.put( host, hostId );
+                {// host not exists, create it.
+
+                    zabbixApi.hostCreate( host, hostGroupCache.get( hostGroup ), ip );
+                    hostCache.put( host, hostid );
                 }
+                return hostid;
             }
         }
         finally
         {
             this.destroy();
         }
-    }
-
-    private String createHost( String host, String ip )
-    {
-        // host not exists, create it.
-        JSONArray groups = new JSONArray();
-        JSONObject group = new JSONObject();
-        group.put( "groupid", hostGroupCache.get( hostGroup ) );
-        groups.add( group );
-
-        // "interfaces": [
-        // {
-        // "type": 1,
-        // "main": 1,
-        // "useip": 1,
-        // "ip": "192.168.3.1",
-        // "dns": "",
-        // "port": "10050"
-        // }
-        // ],
-
-        JSONObject interface1 = new JSONObject();
-        interface1.put( "type", 1 );
-        interface1.put( "main", 1 );
-        interface1.put( "useip", 1 );
-        interface1.put( "ip", ip );
-        interface1.put( "dns", "" );
-        interface1.put( "port", "10051" );
-
-        Request request = RequestBuilder.newBuilder()
-                                        .method( "host.create" )
-                                        .paramEntry( "host", host )
-                                        .paramEntry( "groups", groups )
-                                        .paramEntry( "interfaces", new Object[] { interface1 } )
-                                        .build();
-        JSONObject response = zabbixApi.call( request );
-        return response.getJSONObject( "result" ).getJSONArray( "hostids" ).getString( 0 );
-    }
-
-    private JSONArray getHost( String host )
-    {
-        JSONObject filter = new JSONObject();
-        filter.put( "host", new String[] { host } );
-        Request getRequest = RequestBuilder.newBuilder().method( "host.get" ).paramEntry( "filter", filter ).build();
-        JSONObject getResponse = zabbixApi.call( getRequest );
-        return getResponse.getJSONArray( "result" );
+        return null;
     }
 
     private String itemCacheKey( String host, String item )
     {
         return host + ":" + item;
     }
 
-    void checkItem( String host, String item, int vauleType )
+    String checkItem( String host, String item, int valueType )
     {
+
         try
         {
             if ( itemCache.get( itemCacheKey( host, item ) ) == null )
             {
                 this.zabbixApiInit();
-                JSONArray result = getItem( host, item );
-                if ( result.isEmpty() )
+
```

Comment:
It seems like you shouldn't need to reinitialize the zabbix api component each time, or call destroy() each time. I'm not sure it makes a big difference, but you should be able to manage state at a lower level using an @Injected IndyHttpProvider instance in IndyZabbixApi.

Then, just use the `provider.createClient()` in the `call()` method and make sure you call `finally{ IOUtils.closeQuietly(client);}` before you leave the method. This should manage the state to optimize connection reuse without creating race conditions or the chance of data corruption...it's what we're using in all other parts of Indy now.